internal/driver/drv_sysfs.go

```diff
@@ -0,0 +1,147 @@
+// +build sysfs
+
+package driver
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/spf13/cast"
+)
+
+type GPIODev struct {
+	lc   logger.LoggingClient
+	gpio uint8
+}
+
+func NewGPIODev(lc logger.LoggingClient) *GPIODev {
+	return &GPIODev{lc: lc, gpio: 0}
+}
+
+func (dev *GPIODev) SetGPIO(gpio string, value bool) error {
+	if !strings.Contains(gpio, "/") {
+		return errors.New("invalid gpio number")
+	}
+	split := strings.Split(gpio, "/")
+	valid_port, err := cast.ToUint8E(split[1])
+	if err != nil {
+		return err
+	}
+	if err := export(valid_port); err != nil {
+		return err
+	}
+	if err = setDirection(valid_port, "out"); err != nil {
+		return err
+	}
+	return setValue(valid_port, value)
+}
+
+func (dev *GPIODev) GetGPIO(gpio string) (bool, error) {
+	if !strings.Contains(gpio, "/") {
+		return false, errors.New("invalid gpio number")
+	}
+	split := strings.Split(gpio, "/")
+	valid_port, err := cast.ToUint8E(split[1])
+	if err != nil {
+		return false, err
+	}
+	if err := export(valid_port); err != nil {
+		return false, err
+	}
+	if err = setDirection(valid_port, "in"); err != nil {
+		return false, err
+	}
+	return getValue(valid_port)
+}
+
+func export(gpioNum uint8) error {
+	path := fmt.Sprintf("/sys/class/gpio/gpio%d", gpioNum)
+	if _, err := os.Stat(path); !os.IsNotExist(err) {
+		return nil
+	}
+	return ioutil.WriteFile("/sys/class/gpio/export", []byte(fmt.Sprintf("%d\n", gpioNum)), 0644)
+}
+
+func unexport(gpioNum uint8) error {
```

Comment:
This function doesn't look like it gets used anywhere. Since GOTAGS is set to "chardev", this doesn't get picked up by the compiler. Ideally the service should keep track of all the GPIO pins its exported and unexport them all on clean shutdown.